src/fty_common_messagebus_malamute.cc

```diff
@@ -313,11 +323,17 @@ namespace messagebus {
         if( syncResponse == false ) {
             auto iterator = m_subscriptions.find (subject);
             if (iterator != m_subscriptions.end ()) {
-                //iterator->second(msg);
-                std::thread (iterator->second, msg).detach();
+                try {
+                    (iterator->second)(msg);
+                }
```

Comment:
The reason why we spawned another thread is because we don't want to block message reception if the client has a long-duration callback (for example, making a synchronous request). I'm not a fan of turning this part into a potential huge serialization bottleneck. What we can do alternatively is a worker pool in order not to spawn a thread on each received message.

What do other libraries do in this situation?